apollo-core/src/main/java/com/spotify/apollo/core/ServiceImpl.java

```diff
@@ -208,9 +203,15 @@ private InstanceImpl initInstance(
 
     for (Key<?> key : keysToLoad) {
       Object obj = injector.getInstance(key);
-      if (Closeable.class.isAssignableFrom(obj.getClass())) {
+      if (AutoCloseable.class.isAssignableFrom(obj.getClass())) {
         LOG.info("Managing lifecycle of {}", key.getTypeLiteral());
-        closer.register(Closeable.class.cast(obj));
+        closer.register(() -> {
+          try {
+            AutoCloseable.class.cast(obj).close();
+          } catch (final Exception ex) {
```

Comment:
I'm a bit concerned with suppressing all `Exception`s. As `AutoClosable` javadoc says [https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html](here):

> Implementers of this interface are also strongly advised to not have the close method throw InterruptedException. This exception interacts with a thread's interrupted status, and runtime misbehavior is likely to occur if an InterruptedException is suppressed.

Reply:
That documentation implies that this catch would never get an `InterruptedException` tho. Otherwise I agree

Reply:
I think you could avoid this worry if the original `if closeable, closer.register((Closer)obj)` block was kept and this AutoCloseable block was added as a fallback.

Reply:
I don't really see the reason to treat them differently? One simply extends the other and we can call the super

Reply:
my thinking was that in this case you are handling the exception itself, and in the previous state the closer is handling the exceptions and bubbling them up when `closer.close()` is called, but I am not sure if anyone or any part of Apollo is relying on that.

Reply:
The `Closer` catches `Throwable` anything...